lib/broadway/processor.ex

```diff
@@ -112,6 +114,31 @@ defmodule Broadway.Processor do
     :telemetry.execute([:broadway, :processor, :stop], measurements, metadata)
   end
 
+  defp maybe_prepare_messages(messages, state) do
+    %{module: module, context: context, batchers: batchers} = state
+
+    if function_exported?(module, :prepare_messages, 2) and messages != [] do
```

Comment:
```suggestion
    if function_exported?(module, :prepare_messages, 2) do
```

Messages are always non-empty. :)